### DISCUSSION TITLE
TTGo T4

### DISCUSSION THREAD

**Discussion by gcoego** [micropython/micropython#11921] [ESP32]
Escuse me, its my first time with ESP devices, I bougth my TTGO T4 (4.7" screen and three user buttos) and I don't know where can I download firmware... I've tried with M5Stack with bad results.. somebody help me?  I need a tutorial or quickstart paper...

**Comment by 2dof** [comment]
Did You try  instal with instruction from [[TTGO-T4-Watcher-BTC-ESP32-4-MB-SPI-Flash-4-MB-Psram-display-320-240-TFT](https://github.com/AlexanderSemenov1/TTGO-T4-Watcher-BTC-ESP32-4-MB-SPI-Flash-4-MB-Psram-display-320-240-TFT/tree/master) ? there is   instruction 
``` Инструкция на русском.txt ``` (in Russian) maybe  will help (with google translate) .

**Comment by peterhinch** [comment]
[nano-gui](https://github.com/peterhinch/micropython-nano-gui) supports ILI9341 displays.

**Comment by lbuque** [comment]
T5-4.7？

**Comment by lbuque** [comment]
Here are some micropython firmwares for TTGO products.

https://github.com/lbuque/micropython-builder/actions/runs/5497472810
